packages/common/src/util/http.js

```diff
@@ -195,7 +196,14 @@ export async function request(method, fullUrlOrPath, options = {}) {
 
   await assertOK(response, errors, url, method, startTime);
 
-  const responseBody = await readResponseBody(response, parseAs);
+  const responseBody = await readResponseBody(response, parseAs).catch(() => {
+    throwError(response.statusCode, {
```

Comment:
I was expecting this error handling to inside `readResponseBody`.

Remember `readResponseBody` is called in two places. We don't want to duplicate the error handling.